source/slang/slang-ir-operator-shift-overflow.cpp

```diff
@@ -0,0 +1,66 @@
+// slang-ir-operator-shift-overflow.cpp
+#include "slang-ir-operator-shift-overflow.h"
+
+#include "../../slang.h"
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-layout.h"
+
+namespace Slang {
+
+    class DiagnosticSink;
+    struct IRModule;
+
+    void checkForOperatorShiftOverflowRecursive(
+        IRInst* inst,
+        CompilerOptionSet& optionSet,
+        DiagnosticSink* sink)
+    {
+        if (auto code = as<IRGlobalValueWithCode>(inst))
+        {
+            for (auto block : code->getBlocks())
+            {
+                for (auto opShiftLeft : block->getChildren())
+                {
+                    if (opShiftLeft->getOp() != kIROp_Lsh)
+                        continue;
+
+                    SLANG_ASSERT(opShiftLeft->getOperandCount() == 2);
+
+                    IRInst* rhs = opShiftLeft->getOperand(1);
+                    auto rhsLit = as<IRIntLit>(rhs);
+                    if (!rhsLit)
+                        continue;
+
+                    IRInst* lhs = opShiftLeft->getOperand(0);
+                    IRType* lhsType = lhs->getDataType();
+
+                    IRSizeAndAlignment sizeAlignment;
+                    if (SLANG_FAILED(getNaturalSizeAndAlignment(optionSet, lhsType, &sizeAlignment)))
+                        continue;
+
+                    IRIntegerValue shiftAmount = rhsLit->getValue();
+                    if (sizeAlignment.size * 8 <= shiftAmount)
```

Comment:
instead of using a magic number 8, you may choose to define it as a const / compile-time macro.

Reply:
I think 8 is pretty straight forward.
There is absolutely zero chance that the number 8 will be changed to other numbers like 7 or 9.
Using a constant variable for this seems an unnecessary overkill with less readability.